platform_api/orchestrator/kube_client.py

```diff
@@ -80,6 +90,22 @@ def _raise_status_job_exception(pod: dict[str, Any], job_id: Optional[str]) -> N
         raise JobError("unexpected")
 
 
+def _reraise_as_job_exceptions(
```

Comment:
in general this is fine, but decorators make it more difficult to maintain and debug code. context managers are more explicit and flexible.

Reply:
Good suggestion, please take a look here.